src/analysis/retail/evoker/preservation/modules/talents/DreamBreath.tsx

```diff
@@ -1,39 +1,83 @@
 import SPELLS from 'common/SPELLS';
 import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
-import Events, { ApplyBuffEvent } from 'parser/core/Events';
+import Events, { ApplyBuffEvent, EmpowerEndEvent } from 'parser/core/Events';
 import { ThresholdStyle, When } from 'parser/core/ParseResults';
 import { t } from '@lingui/macro';
 import { SpellLink } from 'interface';
 import { TALENTS_EVOKER } from 'common/TALENTS';
 import { getBuffEvents } from '../../normalizers/CastLinkNormalizer';
+import { RoundedPanel } from 'interface/guide/components/GuideDivs';
+import { GUIDE_CORE_EXPLANATION_PERCENT, GuideContainer } from '../../Guide';
+import { explanationAndDataSubsection } from 'interface/guide/components/ExplanationRow';
+import CastEfficiencyBar from 'parser/ui/CastEfficiencyBar';
+import { GapHighlight } from 'parser/ui/CooldownBar';
+import { BoxRowEntry, PerformanceBoxRow } from 'interface/guide/components/PerformanceBoxRow';
+import { QualitativePerformance } from 'parser/ui/QualitativePerformance';
+
+interface CastInfo {
+  timestamp: number;
+  coyActive: boolean;
+  targetsHit: number;
+  counted: boolean;
+}
 
 class DreamBreath extends Analyzer {
   totalBreaths: number = 0;
   totalApplications: number = 0;
   processedEvents: Set<ApplyBuffEvent> = new Set<ApplyBuffEvent>();
+  casts: CastInfo[] = [];
 
   constructor(options: Options) {
     super(options);
     this.addEventListener(
       Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.DREAM_BREATH),
       this.onApply,
     );
+    // an empowerend will not be generated if using tip the scales
+    this.addEventListener(
+      Events.empowerEnd.by(SELECTED_PLAYER).spell(TALENTS_EVOKER.DREAM_BREATH_TALENT),
+      this.onCast,
+    );
+  }
+
+  onCast(event: EmpowerEndEvent) {
+    this.totalBreaths += 1;
+    const info = {
+      timestamp: event.timestamp,
+      coyActive: this.selectedCombatant.hasBuff(SPELLS.CALL_OF_YSERA_BUFF.id),
+      targetsHit: 0,
+      counted: false,
+    };
+    // ignore cancelled casts
+    if (this.casts.at(-1)?.targetsHit === 0) {
```

Comment:
EmpowerEnd is an extension of BaseCast so it should have an isCancelled property you could check

Reply:
It extends basecastevent so it doesn't have that property, but turned out I didn't need this check after switching from CastEvent to EmpowerEnd.